torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -105,6 +105,10 @@ class TORCH_API Reducer {
   // Cannot combine with the call of `register_comm_hook`.
   void register_builtin_comm_hook(c10d::BuiltinCommHookType comm_hook_type);
 
+  // Runs allreduce or installed communication hook given GradBucket instance.
+  c10::intrusive_ptr<c10::ivalue::Future> run_reduction_hook(
```

Comment:
I think `run_comm_hook` is a better name. Please don't introduce a new concept of reduction hook.